api/src/main/java/org/openmrs/module/eptsreports/reporting/calculation/prev/CompletedIsoniazidProphylaticTreatmentCalculation.java

```diff
@@ -149,30 +150,113 @@ public CalculationResultMap evaluate(
             EptsCalculationUtils.resultForPatient(completedDrugsObservations, patientId);
         Date startDate = getDateFromObs(startProfilaxiaObs);
         Date endDate = getDateFromObs(endProfilaxiaObs);
+
+        // this is because user can type startdate after enddate
         boolean inconsistent =
             (startDate != null && endDate != null && startDate.compareTo(endDate) > 0)
                 || (startDate == null && endDate != null);
-        if ((!inconsistent && startDate != null) || startDrugsObs != null) {
-          boolean completed = startDate != null && endDate != null;
-          if (completed || endDrugsObs != null) {
-            int profilaxiaDuration =
-                Days.daysIn(new Interval(startDate.getTime(), endDate.getTime())).getDays();
-            if (profilaxiaDuration >= MINIMUM_DURATION_IN_DAYS) {
-              map.put(patientId, new BooleanResult(true, this));
-            }
+
+        Date earliestDate = null;
+        Date latestDate = null;
+
+        if ((startDate != null && startDrugsObs == null)
+            && (endDate != null && endDrugsObs == null)) {
+          earliestDate = startDate;
+          latestDate = endDate;
+        }
+
+        if ((startDate != null && startDrugsObs == null)
+            && (endDate == null && endDrugsObs != null)) {
+          earliestDate = startDate;
+          latestDate = endDrugsObs.getEncounter().getEncounterDatetime();
+        }
+        if ((startDate == null && startDrugsObs != null)
+            && (endDate != null && endDrugsObs == null)) {
+          earliestDate = startDrugsObs.getEncounter().getEncounterDatetime();
+          latestDate = endDate;
+        }
+
+        if ((startDate == null && startDrugsObs != null)
```

Comment:
@cfaife @psmatsinhe It would probably be better to isolate the logic here in a method. Also, avoid using the series of if statements as it makes the code complex.

Reply:
done